### DISCUSSION TITLE
Push-to-talk direct audio input unmute and mute

### DISCUSSION THREAD

**Discussion by AtmanActive** [Adam777Z/xbox-controller-button-remapper#18] [Ideas]
Hi,

I'm using this app to turn my 'screenshot' xbox controller button into a push-to-talk button, since I prefer playing online co-op games on my speakers, without a headset.
Currently, Xbox Controller Button Remapper (XCBR)[this app] is set to remap to F22, while another app, [Talkie](https://github.com/alexandrvicente/talkie) is responding to keypress F22 and unmuting/muting the mic.

It is all working very well.

But there is a problem: in games that automatically detect which input was used last (mouse+keyboard vs. gamepad), and change the on-screen prompts accordingly, this approach leads to game UI temporary glitch/confusion. As I am playing on the gamepad, the game is showing, for example, (A) to jump, (B) to crouch, but, when I press the 'screenshot' button on my gamepad, to say something to my co-op buddies, that gets translated to F22, and the game changes the display to show, for example, (space) to jump, (ctrl) to crouch. In other words, the game UI switches to keyboard+mouse mode since the game detects the last input as keypress (F22). A few seconds later, when I continue playing without talking, the game UI switches back to gamepad mode. This is very confusing and very annoying since I need to train my brain to ignore game UI element changes while I'm talking.

There is no workaround for this.

The only possible solution would be to expand XCBR's functionality to support direct windows default audio input unmute/mute. That way there wouldn't be any keypress intermediary and games wouldn't get confused about which input mode prompts to show on screen.

I was looking at [this repository](https://github.com/fredemmott/StreamDeck-AudioMute) source code as that one executes the task perfectly. In any case, any simple C++ code to mute/unmute would do.

Thanks.

**Comment by Adam777Z** [comment]
You are correct.

Added in version [1.9.1.0](https://github.com/Adam777Z/xbox-controller-button-remapper/releases/tag/1.9.1.0).

**Comment by AtmanActive** [comment]
This is great news!
Thank you very much for this.

It is working correctly here.

Just one caveat though:
By default, config.ini has the setting 'share_hold_mode' set to 2.
But, when I set 'share_key' to 905, expecting audio push-to-talk functionality, without touching anything else in the config file, my first experience was the app crashing as soon as I press the share key on the controller.
Then, by examining my old config versus this new config, line by line, I saw that my old config had 'share_hold_mode = 1'. So, when I applied this same line to the new config, it all started working as expected.
